app/tests/integration/nimbus/test_e2e_create_experiment.py

```diff
@@ -1,57 +1,81 @@
 import time
 
 import pytest
+from nimbus.models.base_dataclass import (
+    BaseAudienceDataClass,
+    BaseBranchDataClass,
+    BaseDataClass,
+)
 from nimbus.pages.home import HomePage
 from nimbus.pages.summary import SummaryPage
 from nimbus.remote_settings.pages.dashboard import Dashboard
 from nimbus.remote_settings.pages.login import Login
 from selenium.common.exceptions import NoSuchElementException, TimeoutException
 
 
-@pytest.mark.nondestructive
-def test_create_new_experiment(selenium, base_url):
-    experiment_name = "test_create_new_experiment"
-
-    selenium.get(base_url)
-    home = HomePage(selenium, base_url).wait_for_page_to_load()
-    experiment = home.create_new_button()
-    experiment.public_name = experiment_name
-    experiment.hypothesis = "smart stuff here"
-    experiment.application = "DESKTOP"
+@pytest.fixture(name="default_data", scope="module")
+def fixture_default_data():
+    return BaseDataClass(
+        hypothesis="smart stuff here",
+        application="DESKTOP",
+        public_description="description stuff",
+        branches=[
+            BaseBranchDataClass(
+                name="name 1",
+                description="a nice experiment",
+                config="No Feature Firefox Desktop",
+            )
+        ],
+        audience=BaseAudienceDataClass(
+            channel="Nightly",
+            min_version=80,
+            targeting="TARGETING_MAC_ONLY",
+            percentage=50.0,
+            expected_clients=50,
+        ),
+    )
+
+
+def create_experiment(selenium, home_page, data):
```

Comment:
This could be a fixture and moved to ```conftest.py```.

Reply:
The problem is that the experiment name of each test has to be different to make sure there are no conflicts between tests. I saw that there is a way of [parameterizing fixtures using pytest annotations](https://docs.pytest.org/en/latest/example/parametrize.html), but it looks a bit hacky to me. Should I still use it to define the experiment name for each test or do you have another idea?

Reply:
Maybe concatenate some random numbers or a shortened UUID on the end of the assignment. 

```suggestion
data.public_name + UUID generation or something
```

Reply:
Do you still plan to move this to ```conftest.py```?

Reply:
I do! Will take some time tomorrow to move this into a fixture and implement your random number / UUID idea!

Reply:
I liked having the experiment name be the name of the test so it's easier to debug the tests after the fact?

Reply:
@jaredlockhart I agree. In case we actually go for the generated experiment names, I think we would at least have to log them.

@jrbenny35 I just started putting the `create_experiment` function into a fixture. I think this might not be as easy as it seems. In some tests, we have steps that have to be executed **before** the creation of the experiment, e.g. counting the number of experiments in the `test_create_new_experiment_remote_settings` test or navigating to the Drafts tab in `test_create_new_experiment_remote_settings_reject`. Probably, in order to keep this behavior, we would need to move those parts of the code into fixtures as well. If the only reason for putting this function into a fixture is to make it available in multiple test files, I would suggest we don't create a fixture that executes `create_experiment`, but instead returns the `create_experiment` function itself, so we can execute it whenever we want. This pattern is also explained in the [pytest docs](https://docs.pytest.org/en/latest/how-to/fixtures.html#factories-as-fixtures). What do you think?

Reply:
@lennartschoch I like that. Let's do that.